lib/bap_types/bap_helpers.ml

```diff
@@ -91,9 +91,7 @@ class substitution x y = object(self)
   method! map_let z ~exp ~body =
     if Bap_var.(z = x)
     then Let (z,self#map_exp exp,body)
-    else super#map_let z
-        ~exp:(self#map_exp exp)
-        ~body:(self#map_exp body)
+    else super#map_let z ~exp ~body
```

Comment:
This looks right to me, but, the other change that was previously present in this file was a rogue reversion so someone should confirm.

Reply:
I am confirming this change.

The other one wasn't really rogue (in a sense that it is not git that we should blame, but it was me playing with the code and accidentally reverting to polymorphic compare, which wasn't banned at that times).